From 5d4727a5133bfd5d166d15c53f68a2cb854fad38 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Wed, 8 Oct 2014 15:44:24 +0200
Subject: [PATCH 1/4] bcmdhd: add BCM4324/4339 KConfig and Makefile

Add support for KConfig and Makefile configuration for
BCM4324/4339 chip families

Change-Id: If04b499292daa1c7689b6f9587180ec200f86783
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 drivers/net/wireless/bcmdhd/Kconfig  |   11 +++++++++++
 drivers/net/wireless/bcmdhd/Makefile |   11 +++++++++++
 2 files changed, 22 insertions(+)

diff --git a/drivers/net/wireless/bcmdhd/Kconfig b/drivers/net/wireless/bcmdhd/Kconfig
index e3a3b6a809f1..bd4ea08923be 100644
--- a/drivers/net/wireless/bcmdhd/Kconfig
+++ b/drivers/net/wireless/bcmdhd/Kconfig
@@ -15,9 +15,20 @@ config BCMDHD_PCIE
 	bool "PCIe bus interface support"
 	depends on BCMDHD && PCI && !BCMDHD_SDIO
 
+config BCM4324
+	tristate "BCM4324 support"
+	depends on BCMDHD
+	default y
+
+config BCM4339
+	tristate "BCM4339 support"
+	depends on BCMDHD
+	default n
+
 config BCM4354
 	tristate "BCM4354 support"
 	depends on BCMDHD
+	default n
 
 config BCM4356
 	tristate "BCM4356 support"
diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index a1ed1660b72c..7f2acf2833ce 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -190,6 +190,17 @@ ifneq ($(CONFIG_BCM4339),)
   DHDCFLAGS += -DCUSTOM_PSPRETEND_THR=30
 endif
 
+ifneq ($(CONFIG_BCM4324),)
+  DHDCFLAGS += -DBDC -DOOB_INTR_ONLY -DDHD_BCMEVENTS -DMMC_SDIO_ABORT
+  DHDCFLAGS += -DBCMSDIO -DBCMLXSDMMC -DUSE_SDIOFIFO_IOVAR
+  DHDCFLAGS += -DPROP_TXSTATUS -DPROP_TXSTATUS_VSDB
+
+  # tput enhancement
+  DHDCFLAGS += -DHW_OOB
+  DHDCFLAGS += -DCUSTOM_GLOM_SETTING=5
+  DHDCFLAGS += -DCUSTOM_SDIO_F2_BLKSIZE=128
+endif
+
 
 #EXTRA_LDFLAGS += --strip-debug
 
-- 
1.7.9.5

